src/libraries/System.Console/src/System/IO/StdInReader.cs

```diff
@@ -46,11 +46,15 @@ internal bool IsUnprocessedBufferEmpty()
 
         internal void AppendExtraBuffer(ReadOnlySpan<byte> buffer)
         {
-            // Ensure a reasonable upper bound applies to the stackalloc
-            Debug.Assert(buffer.Length <= 1024);
-
-            // Then convert the bytes to chars
-            Span<char> chars = stackalloc char[_encoding.GetMaxCharCount(buffer.Length)];
+            // Most inputs to this will have a buffer length of one.
+            // The cases where it is larger than one only occur in ReadKey
+            // when the input is not redirected, so those cases should be
+            // rare, so just allocate.
+            const int MaxStackAllocation = 256;
+            int maxCharsCount = _encoding.GetMaxCharCount(buffer.Length);
+            Span<char> chars = (uint)maxCharsCount <= MaxStackAllocation ?
```

Comment:
just curious, why is the cast to uint required?

Reply:
To simultaneously check if `maxCharsCount` is not negative. If it was, its unsigned form would be very big, and the condition would fail. Whether `GetMaxCharCount` can actually be negative is another story; I guess the cast is to guard against 3rd-party `Encoding` descendants.

Reply:
Ah, the derived classes was what I didn't think of.